files/internals/cports.common

```diff
@@ -189,14 +373,50 @@ fi
 
 if [ "$EGF" == "1" ]; then
  if [ "$EG_DROP_CMD" == "1" ]; then
-		$IPT $IPT_FLAGS -N DEG
+	if [ $EXECUTOR == "nft" ]; then
+           if [ "$USE_IPV6" == "1" ]; then
+              $NFT add chain ip6 filter DEG
+           else
+	      $NFT add chain ip filter DEG
+	   fi
+	else
+           $IPT $IPT_FLAGS -N DEG
+	fi
 	for i in `echo $EG_DROP_CMD | tr ',' ' '`; do
 		si=`echo $i | cut -c 1-6`		
 		if [ "LOG_DROP" == "1" ]; then 
-		  $IPT $IPT_FLAGS -A DEG -m limit --limit=$LOG_RATE/minute -j $LOG_TARGET --log-level=$LOG_LEVEL $LEXT --log-prefix="** DEG_$si ** "
+		   if [ $EXECUTOR == "nft" ]; then
+		      if [ "$USE_IPV6" == "1" ]; then
+		         $NFT add rule ip6 filter DEG limit rate $LOG_RATE/minute burst 5 packets counter log prefix \"** DEG_$si ** \" level $LOG_LEVEL $LEXT
+		      else
+                         #$IPT $IPT_FLAGS -A DEG -m limit --limit=$LOG_RATE/minute -j $LOG_TARGET --log-level=$LOG_LEVEL $LEXT --log-prefix="** DEG_$si ** "
+		         $NFT add rule ip filter DEG limit rate $LOG_RATE/minute burst 5 packets counter log prefix \"** DEG_$si ** \" level $LOG_LEVEL $LEXT
+		      fi
+                   else
+                      $IPT $IPT_FLAGS -A DEG -m limit --limit=$LOG_RATE/minute -j $LOG_TARGET --log-level=$LOG_LEVEL $LEXT --log-prefix="** DEG_$si ** "
+		   fi
+		fi
+		if [ $EXECUTOR == "nft" ]; then
+                   # --cmd-owner is removed in kernel 2.6.15 https://superuser.com/questions/34782/with-linux-iptables-is-it-possible-to-log-the-process-command-name-that-initiat
+                   # https://android.googlesource.com/platform/external/iptables.git/+/eclair-sholes-release2/extensions/libipt_owner.c
+                   # https://www.spinics.net/lists/netfilter/msg49715.html
+                   #
+                   # TODO is this good? or maybe eout said not supported?
+                   # because this match socket uid than command owner id.
+                   #
+                   if [ "$USE_IPV6" == "1" ]; then
+                      $NFT add rule ip6 filter DEG skuid $i counter jump $ALL_STOP
+                   else
+                      $NFT add rule ip filter DEG skuid $i counter jump $ALL_STOP
+                   fi
+                else
+                   $IPT $IPT_FLAGS -A DEG -s 0/0 -d 0/0 -m owner --cmd-owner=$i -j $ALL_STOP
 		fi
-		$IPT $IPT_FLAGS -A DEG -s 0/0 -d 0/0 -m owner --cmd-owner=$i -j $ALL_STOP
 	done
-		  $IPT $IPT_FLAGS -A OUTPUT -j DEG
+                if [ $EXECUTOR == "nft" ]; then
+		   $NFT add rule ip filter OUTPUT counter jump DEG
+                else
+		   $IPT $IPT_FLAGS -A OUTPUT -j DEG
+		fi
```

Comment:
put into an outer nft if clause easier? cuz now in and out of the if clauses make me dizzy.

Reply:
In `nftables`, the `meta` module can match on a variety of packet metadata, including the process command name, or "comm", using the `comm` field.

The `comm` field allows you to match packets based on the command name of the process that generated them. This is useful for filtering traffic generated by specific applications or services.

Here's an example of how to match packets based on the command name using the `comm` field in `nftables`:

```
nft add rule ip filter input meta comm eq "ssh" drop
```
This rule drops all incoming packets whose process command name is "ssh". You can replace `"ssh"` with the name of any other process to match packets generated by a different application or service.

You can also use other comparison operators, such as `neq` (not equal) or `contains` (matches a substring), with the `comm` field to refine your packet filtering criteria.

Keep in mind that the `comm` field is based on the value of the `argv[0]` field in the process's `task_struct` structure, which can be changed by the process itself. As a result, using the `comm` field for security purposes is not completely reliable.